app/views/recipe_foods/new.html.erb

```diff
@@ -0,0 +1,22 @@
+<h4>Add Recipe Food</h4>
+<%= form_with model: [@recipe, @recipe.recipe_foods.create] do |form|%>
+<div class="mb-3">
+  <%= form.label :food %>
+  <div>
+    <%= form.select :food_id, current_user.foods.collect {|food| [food.name,
+    food.id]},{ include_blank: true} , class:"form-select" %>
```

Comment:
- [ ] Kindly the user should not only select his own food rather he should get access to all food in the database and select one. 
```html.rb
 <%= form.select :food_id, Food.all.collect {|food| [food.name,
    food.id]},{ include_blank: true} , class:"form-select" %>
```
or set @foods in your controller `recipe_foods#new` and access it in your view.
```rb
  def new
    @recipe = Recipe.find(params[:recipe_id])
    @recipe_food = @recipe.recipe_foods.new
    @foods = Food.all
  end
```